File: Recursion_Array/firstIndex.java
import java.util.Scanner;
public class firstIndex
{
    public static Scanner scn = new Scanner(System.in);
    public static int first(int[] arr, int idx, int data)
    {
        if(idx == arr.size())
        {
            return -1;
        }
        if(arr[idx] == data)
        {
            return idx;
        }
        return first(arr, idx + 1, data);
    }

    public static void input(int[] arr)
    {
        for(int i = 0; i < arr.length; i++)
        {
            arr[i] = scn.nextInt();
        }
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        int data = scn.nextInt();
        System.out.println(first(arr, 0, data));
    }
}

File: Recursion_Array/firstIndex.cpp
#include<iostream>
#include<vector>
using namespace std;

int first(vector<int>& arr, int idx, int data)
{
    if(idx == arr.size())
    {
        return -1;
    }
    if(arr[idx] == data)
    {
        return idx;
    }
    return first(arr, idx + 1, data);
}

void input(vector<int>& arr)
{
    for(int i = 0; i < arr.size(); i++)
    {
        cin >> arr[i];
    }
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    input(arr);
    int data;
    cin >> data;
    cout << first(arr, 0, data);
}